pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -170,7 +170,11 @@ protected void handleConnect(CommandConnect connect) {
             // Client is doing a lookup, we can consider the handshake complete and we'll take care of just topics and
             // partitions metadata lookups
             state = State.ProxyLookupRequests;
-            lookupProxyHandler = new LookupProxyHandler(service, this);
+            if (service.getConfiguration().isDiscoveryServiceEnabled()) {
```

Comment:
is mentioned above, should we derive it based on global-zk value present or not?